[Bug]: hcloud plugin crashed

### What happened?

### Description

```
hcloud_volume.chatty_db: Creating...
module.k3s.hcloud_load_balancer_service.control_plane[0]: Creating...
hcloud_zone_rrset.chatty_apex_soa: Creating...
module.k3s.hcloud_load_balancer_network.control_plane[0]: Creating...
hcloud_zone_rrset.chatty_apex_ns: Creating...
module.k3s.module.control_planes["0-0-control-plane-ash"].hcloud_server.server: Creating...
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.
╵
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
╷
│ Error: API request failed
│ 
│   with hcloud_zone_rrset.chatty_apex_ns,
│   on dns.tf line 24, in resource "hcloud_zone_rrset" "chatty_apex_ns":
│   24: resource "hcloud_zone_rrset" "chatty_apex_ns" {
│ 
│ An unexpected error was encountered during an API request.
│ 
│ RRSet already exists
│ 
│ Error code: uniqueness_error
│ Status code: 409
│ 
╵
╷
│ Error: name is already in use (uniqueness_error, afc6ca59576b39489f2c94aef994c26e)
│ 
│   with hcloud_volume.chatty_db,
│   on storage.tf line 1, in resource "hcloud_volume" "chatty_db":
│    1: resource "hcloud_volume" "chatty_db" {
│ 
╵

Stack trace from the terraform-provider-hcloud_v1.57.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd53e44]

goroutine 130 [running]:
github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil.SettleActions.func2.1(0xc000680530?)
        github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil/actions.go:37 +0x4
slices.IndexFunc[...](...)
        slices/slices.go:109
slices.DeleteFunc[...]({0xc0005405a0?, 0x2, 0x2}, 0xc0008f69c8)
        slices/slices.go:237 +0x5e
github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil.SettleActions.func2(0xc0000b9360)
        github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil/actions.go:36 +0xa5
github.com/hetznercloud/hcloud-go/v2/hcloud.(*ActionClient).WaitForFunc(0xc000343238, {0x135eb38, 0xc00095b290}, 0xc00050fd58, {0xc000540590?, 0xee0efc88e?, 0x0?})
        github.com/hetznercloud/hcloud-go/v2@v2.32.0/hcloud/action_waiter.go:35 +0x132
github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil.SettleActions({0x135eb38, 0xc00095b290}, {0x1355d60, 0xc000343238}, {0xc000540590, 0x2, 0x2})
        github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil/actions.go:33 +0x1ba
github.com/hetznercloud/terraform-provider-hcloud/internal/zonerrset.(*Resource).Create(0xc000a02038, {0x135eb38, 0xc00095b290}, {{{{0x1365680, 0xc0009a6630}, {0x1023f80, 0xc000a1e660}}, {0x1367ad0, 0xc0002b0690}}, {{{0x1365680, ...}, ...}, ...}, ...}, ...)
        github.com/hetznercloud/terraform-provider-hcloud/internal/zonerrset/resource.go:226 +0xad8
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc0003d0308, {0x135eb38, 0xc00095b290}, 0xc0008f7420, 0xc0008f73f0)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/fwserver/server_createresource.go:127 +0x88e
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc0003d0308, {0x135eb38, 0xc00095b290}, 0xc000232150, 0xc0008f7548)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/fwserver/server_applyresourcechange.go:63 +0x586
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0xc0003d0308, {0x135eb38?, 0xc00095b1a0?}, 0xc000960280)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/proto6server/server_applyresourcechange.go:71 +0x585
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0xc0003cc1b0, {0x135eb38?, 0xc00095aed0?}, 0xc000960280)
        github.com/hashicorp/terraform-plugin-mux@v0.21.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0000b8280, {0x135eb38?, 0xc00095a690?}, 0xc000a08080)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/tf6server/server.go:944 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x11715e0, 0xc0000b8280}, {0x135eb38, 0xc00095a690}, 0xc000a08000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:789 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ac200, {0x135eb38, 0xc00095a600}, 0xc000300c00, 0xc0000bfe00, 0x1b7a9e8, 0x0)
        google.golang.org/grpc@v1.75.1/server.go:1431 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc0002ac200, {0x135f698, 0xc000178000}, 0xc000300c00)
        google.golang.org/grpc@v1.75.1/server.go:1842 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.75.1/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 29
        google.golang.org/grpc@v1.75.1/server.go:1072 +0x11d

Error: The terraform-provider-hcloud_v1.57.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

make: *** [Makefile:369: deploy-cluster] Error 1
```

It seems that problem happens when cluster already has the resource with the same name.

### Kube.tf file

```terraform
private
```

### Screenshots

_No response_

### Platform

Linux Mint Cinnamon 22.2

### What did you expect to happen?

no crash

### Please provide a minimal working example

https://github.com/mysticaltech/terraform-hcloud-kube-hetzner/issues/1992

--- a/internal/util/hcloudutil/actions.go
+++ b/internal/util/hcloudutil/actions.go
@@ -17,14 +17,16 @@ type ActionWaiter interface {
 }
 
 func SettleActions(ctx context.Context, client ActionWaiter, actions ...*hcloud.Action) (diags diag.Diagnostics) {
-	runningActions := make([]*hcloud.Action, len(actions))
-	copy(runningActions, actions)
-	failedActions := make([]*hcloud.Action, 0)
+	// Filter out nil actions
+	actions = slices.DeleteFunc(actions, func(a *hcloud.Action) bool { return a == nil })
+
+	running := slices.Clone(actions)
+	failed := make([]*hcloud.Action, 0)
 
 	// Make sure we always report failed actions, even if an API calls fails for other reasons and we return early
 	defer func() {
-		if len(failedActions) > 0 {
-			for _, action := range failedActions {
+		if len(failed) > 0 {
+			for _, action := range failed {
 				diags.Append(ActionErrorDiagnostic(action))
 			}
 		}
@@ -32,21 +34,21 @@ func SettleActions(ctx context.Context, client ActionWaiter, actions ...*hcloud.
 
 	err := client.WaitForFunc(ctx, func(update *hcloud.Action) error {
 		if update.Status == hcloud.ActionStatusSuccess || update.Status == hcloud.ActionStatusError {
-			// Remove from runningActions
-			runningActions = slices.DeleteFunc(runningActions, func(action *hcloud.Action) bool {
+			// Remove from running
+			running = slices.DeleteFunc(running, func(action *hcloud.Action) bool {
 				return action.ID == update.ID
 			})
 
 			if update.Status == hcloud.ActionStatusError {
-				failedActions = append(failedActions, update)
+				failed = append(failed, update)
 			}
 		}
 
 		return nil
 	}, actions...)
 	if err != nil {
-		if (errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded)) && len(runningActions) > 0 {
-			diags.Append(ActionWaitTimeoutDiagnostic(runningActions...))
+		if (errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded)) && len(running) > 0 {
+			diags.Append(ActionWaitTimeoutDiagnostic(running...))
 			return
 		}
 

